soh/src/overlays/actors/ovl_En_Ma1/z_en_ma1.c

```diff
@@ -290,10 +309,14 @@ void EnMa1_Init(Actor* thisx, GlobalContext* globalCtx) {
     this->actor.targetMode = 6;
     this->unk_1E8.unk_00 = 0;
 
-    if (!(gSaveContext.eventChkInf[1] & 0x10) || (CHECK_QUEST_ITEM(QUEST_SONG_EPONA) && !gSaveContext.n64ddFlag) ||
+    // If Talon has not left Hyrule Castle, we are randomized and ahve not obtained Malon's HC Check, we are not randomized
```

Comment:
Ok, just made a commit to address some of that. Malon's Actor is complicated in general because there's a lot of shared data between her appearances in Hyrule Castle and Lon Lon Ranch, so the conditionals being kind of complicated are unavoidable.